### ISSUE TITLE
[Label] - allow clickable labels to be disabled

### ISSUE THREAD

**Issue by Gkrumbach07** [patternfly/patternfly#6055]
**Is this a new component or an extension of an existing one?**
This is using the Label component with `onClick` defined.

**Describe the feature**
Because labels can now be clickable and act as buttons, they should also be able to have a disabled state with the accompanying styling.

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
[Code sandbox](https://codesandbox.io/s/modest-sanderson-5qpgdt?file=/fonts.css) provided by @mcoker 
![image](https://github.com/patternfly/patternfly/assets/12587674/cc5f56e4-6243-4d76-8e49-79bdc453215a)

**Any other information?**
This was talked about in a slack thread [here](https://patternfly.slack.com/archives/C293LQ36J/p1699429646350539) where we had a list of labels and a clickable label that routed to a new page. In some case that label button needed to be disabled. We didn't want to apply any custom styling so we instead had to use a link variant button to sort of match the non disabled label.

**Comment by wise-king-sullyman** [comment]
Hello @Gkrumbach07, thanks for submitting this issue but I don't think we're quite understanding the use case.

Would a Button be a better fit here? Or put another way, is there a reason that a Button can't be used in place of a clickable label for your use case?

**Comment by Gkrumbach07** [comment]
@wise-king-sullyman We tried a button but it ended up looking too bulky for the table cell we had it in.

This is the UX we have now
![image](https://github.com/patternfly/patternfly/assets/12587674/96ffbcd8-5443-4647-8452-77045be8135e)

The table cell displays a label list with a button to essentially add more to the list. But we wanted the ability to disable this button, and the small version of a button was still not the same font size as the label. Our disabled solution is using a button, but we had to manually make the font size `xs` to match the labels.

Ideally we would like to just be able to disable the label if it is clickable, becuase that was our original design
